src/analysis/validation.jl

```diff
@@ -2,102 +2,238 @@
+    head_wntr = Array{Float64,1}(undef,duration)
+    head_watermodels = Array{Float64,1}(undef,duration)
+    pressure_wntr = Array{Float64,1}(undef,duration)
+    pressure_watermodels = Array{Float64,1}(undef,duration)
     
-    wntr_flow = wslinks["flowrate"]
-    wntr_dh = wslinks["headloss"]
-
-    link_flow_wntr = Dict{String,Any}()
-    link_flow_wm = Dict{String,Any}()
-
-    link_dh_wntr = Dict{String,Any}()   # head gain/loss
-    link_dh_wm = Dict{String,Any}()
+    for t in 1:duration
+        elevation[t] = wntr_data.nodes._data[node_name].elevation
+        head_wntr[t] = wntr_simulation.node["head"][node_name].values[t]
+        head_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["node"][node_id]["h"]
+        pressure_wntr[t] = wntr_simulation.node["pressure"][node_name].values[t]
+        pressure_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["node"][node_id]["p"]
+    end
 
-    for ltype in link_types
-        for (key, link) in solution[string("1")][ltype]
+    node_df = DataFrame(time = 1:time_step:time_step*duration, elevation = vec(elevation), 
+        head_wntr = vec(head_wntr), head_watermodels = vec(head_watermodels),
+        pressure_wntr = vec(pressure_wntr), pressure_watermodels = vec(pressure_watermodels))
 
-            # if ltype in ["check_valve", "shutoff_valve"]
-            #     link_name = data["pipe"][key]["name"]
-            # else
-            #     link_name = data[ltype][key]["name"]
-            # end
+    return node_df
+end
 
-            link_name = data[ltype][key]["name"]
+# compare tanks
+function get_tank_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation, tank_id)
+    duration = length(wm_solution["solution"]["nw"])            # number of time steps
+    time_step = wm_data["time_step"]/3600                       # length per time step (hour)
+    tank_name = wm_data["nw"]["1"]["tank"][tank_id]["source_id"][2]
+    diameter = wntr_data.nodes._data[tank_name].diameter
 
+    volume_wntr = Array{Float64,1}(undef,duration)
+    volume_watermodels = Array{Float64,1}(undef,duration)
+    level_wntr = Array{Float64,1}(undef,duration)
+    level_watermodels = Array{Float64,1}(undef,duration)
 
-            link_flow_wntr[link_name] = Array{Float64,1}(undef,length(solution))
-            link_flow_wm[link_name] = Array{Float64,1}(undef,length(solution))
+    for t in 1:duration
+        volume_wntr[t] = wntr_simulation.node["pressure"][tank_name].values[t]*(1/4*pi*diameter^2)
+        volume_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["tank"][tank_id]["V"]
 
-            link_dh_wntr[link_name] = Array{Float64,1}(undef,length(solution))
-            link_dh_wm[link_name] = Array{Float64,1}(undef,length(solution))
-        end
+        level_wntr[t] = wntr_simulation.node["pressure"][tank_name].values[t]
+        level_watermodels[t] = wm_solution["solution"]["nw"][string(t)]["tank"][tank_id]["V"]/(1/4*pi*diameter^2)
```

Comment:
Again, I'd use `"p"` here at the corresponding tank node to compute the WaterModels level.